### PR TITLE
Send txFee to proposer not to infrastructure fund

### LINKED ISSUE #455: Block proposers SBAT receive a portion of the transaction fee
### Expected Behavior

s/infrastructure_fraction/proposer_fraction

### Current Behavior

What is the current behavior?

### PR DESCRIPTION
**Pull request by mcortesi**
### Description

Instead of sending gas fees to "feeRecipient" and "infraestructureFund" we send them to "feeRecipient" and the "proposer".

The proposer, it's also the coinbase for the node.

### Tested

e2e

### Related issues

- Fixes #455

### CODE REVIEW

**Comment by mcortesi** [comment]
btw @asaj @kevjue 
vscode automatically formatted my code and removes spaces in 
`type StateTransition struct {`
and later when creates an instance of it.

didn't know if that's correct or how to change it

File: contract_comm/gasprice_minimum/gasprice_minimum.go [hunk 1]
```diff
@@ -98,13 +98,13 @@ const (
 const defaultGasAmount = 2000000
 
 var (
-	gasPriceMinimumABI, _                           = abi.JSON(strings.NewReader(gasPriceMinimumABIString))
-	FallbackInfraFraction   *InfrastructureFraction = &InfrastructureFraction{big.NewInt(0), big.NewInt(1)}
-	FallbackGasPriceMinimum *big.Int                = big.NewInt(0) // gasprice min to return if contracts are not found
-	suggestionMultiplier    *big.Int                = big.NewInt(5) // The multiplier that we apply to the minimum when suggesting gas price
+	gasPriceMinimumABI, _                     = abi.JSON(strings.NewReader(gasPriceMinimumABIString))
+	FallbackInfraFraction   *ProposerFraction = &ProposerFraction{big.NewInt(0), big.NewInt(1)}
```
Comment (asaj) [inline contract_comm/gasprice_minimum/gasprice_minimum.go]: s/FallbackInfraFraction/FallbackProposerFraction?

File: core/state_transition.go [hunk 2]
```diff
@@ -370,33 +366,24 @@ func (st *StateTransition) TransitionDb() (ret []byte, usedGas uint64, failed bo
...
 	if txFeeRecipient == nil {
 		sender := msg.From()
 		txFeeRecipient = &sender
 	}
-	err = st.creditGas(*txFeeRecipient, recipientTxFee, msg.GasCurrency())
 
-	if err != nil {
+	gasUsed := new(big.Int).SetUint64(st.gasUsed())
+	totalTxFee := new(big.Int).Mul(gasUsed, st.gasPrice)
+	// gasUsed * gasPriceMinimun * proposerFractionNumerator / proposerFractionDenominator
```
Comment (asaj) [inline core/state_transition.go]: Remove?

**Comment by asaj** [comment]
Is there a corresponding monorepo PR?

File: core/state_transition.go (hunk 2 above)
Comment (mcortesi) [reply core/state_transition.go]: The intent was to clarify what's the math on the line below.
IMO, constantly using `new(big.Int).` makes the lines harder to read.

**Pull request merged**